server/routes/serviceProviderReferrals/showReferralPresenter.ts

```diff
@@ -39,24 +41,41 @@ export default class ShowReferralPresenter {
     { key: 'Email address', lines: [this.sentBy.email ?? ''] },
   ]
 
-  get interventionDetails(): SummaryListItem[] {
-    // const selectedDesiredOutcomes = this.serviceCategory.desiredOutcomes
-    //   .filter(desiredOutcome => this.sentReferral.referral.desiredOutcomesIds.includes(desiredOutcome.id))
-    //   .map(desiredOutcome => desiredOutcome.description)
-    //
-    // const selectedComplexityLevel = this.serviceCategory.complexityLevels.find(
-    //   complexityLevel => complexityLevel.id === this.sentReferral.referral.complexityLevelId
-    // )
-    //
-    // const complexityLevelText = {
-    //   level: selectedComplexityLevel?.title || 'Level not found',
-    //   text: selectedComplexityLevel?.description || 'Description not found',
-    // }
+  get referralServiceCategories(): ServiceCategory[] {
+    const { serviceCategoryIds } = this.sentReferral.referral
+    return this.intervention.serviceCategories.filter(it => serviceCategoryIds.includes(it.id))
+  }
+
+  serviceCategorySection(serviceCategory: ServiceCategory, tagMacro: (args: TagArgs) => string): SummaryListItem[] {
+    const items: SummaryListItem[] = []
+
+    const complexityLevel = this.getReferralComplexityLevelForServiceCategory(serviceCategory)
+    if (complexityLevel === null) {
+      items.push({ key: 'Complexity level', lines: ['Level not found'] })
+    } else {
+      items.push({
+        key: 'Complexity level',
+        lines: [tagMacro(PresenterUtils.complexityLevelTagArgs(complexityLevel)), complexityLevel.description],
+      })
+    }
 
+    items.push({ key: 'Desired outcomes', lines: ['Outcomes not found'] })
+
+    return items
+  }
+
+  private getReferralComplexityLevelForServiceCategory(serviceCategory: ServiceCategory): ComplexityLevel | null {
+    const complexityLevelId = this.sentReferral.referral.complexityLevels.find(
+      it => it.serviceCategoryId === serviceCategory.id
+    )?.complexityLevelId
+
+    return serviceCategory.complexityLevels.find(it => it.id === complexityLevelId) || null
```

Comment:
How likely is it that we'll return `null` here? I've been thinking about how nullable our codebase is recently, and wondering if throwing errors would make it easier to work with? Maybe not, as it makes the page unusable vs rendering an empty field, although would an empty field be useful from a user perspective?

Reply:
yeh i know what you mean. 

like it's totally an error condition if the selected complexity level doesn't exist for the selected service category. i'd rather not kill the page, but we could log an error?